core/src/main/java/org/hibernate/ogm/persister/impl/OgmEntityPersister.java

```diff
@@ -909,20 +911,16 @@ protected void createLoaders() {
 		loaders.put( LockMode.OPTIMISTIC, createEntityLoader( LockMode.OPTIMISTIC ) );
 		loaders.put( LockMode.OPTIMISTIC_FORCE_INCREMENT, createEntityLoader( LockMode.OPTIMISTIC_FORCE_INCREMENT ) );
 
-		//FIXME handle cascading merge and refresh
 		loaders.put(
 				"merge",
-				createEntityLoader( LockMode.READ )
-				//new CascadeEntityLoader( this, CascadingAction.MERGE, getFactory() )
+				new CascadeEntityLoader( this, CascadingActions.MERGE, getFactory() )
 			);
 		loaders.put(
 				"refresh",
-				createEntityLoader( LockMode.READ )
-				//new CascadeEntityLoader( this, CascadingAction.REFRESH, getFactory() )
+				new CascadeEntityLoader( this, CascadingActions.REFRESH, getFactory() )
```

Comment:
With these changes, I think we can remove the whole overriding `createLoaders()` method and use the one from ORM. That was my ultimate goal for this issue as I made significant changes to ORM in this area and I would like these changes to come for free to OGM.

Could you try and check?

Reply:
Very good idea. Remove code it is always better ;) I'm going to check if it really possible

Reply:
@gsmet currently is it not possible to remove the whole method. If we wanted this, we should resize the scope of the issue, adding more changes on it, or alternatively we could open a follow up issue.

Reply:
What's the issue exactly? You're too vague to convince me :).

Reply:
:) nice approach. In the current implementation of Hibernate ORM, *noneLockLoader* and r*eadLockLoader* are moved out from the loaders map. So in order to use the whole base method we have to change the getAppropriateLoader too, leading us out of scope of OGM-1376, from what I think.

Reply:
Hum, yes, you're right, `getLoaderByLockMode()` should have been accessible to the child classes. Bummer.

I opened https://hibernate.atlassian.net/browse/HHH-12615 to change that.

Reply:
nice followup!